tests/integration_tests/build/test_coverage.py

```diff
@@ -32,6 +32,44 @@
 """Regex for extracting number of total executable lines found by kcov."""
 
 
+@pytest.mark.timeout(120)
+@pytest.mark.skipif(
+    platform.machine() != "x86_64",
+    reason="kcov hangs on aarch64"
+)
+def test_ensure_mod_tests():
+    """Check that files containing unit tests have a 'tests' module defined."""
+    # List all source files containing rust #[test] attribute,
+    # (excluding generated files and integration test directories).
+    # Take the list and check each file contains 'mod tests {', output file
+    # name if it doesn't.
+    cmd = (
+        '/bin/bash '
+        '-c '
+        '"grep '
+        '--files-without-match '
+        '\'mod tests {\' '
+        '\\$(grep '
+        '--files-with-matches '
+        '--recursive '
+        '--exclude-dir=src/*_gen/* '
+        '\'\\#\\[test\\]\' ../src/*/src)" '
+    )
+
+    # The outer grep returns 0 even if it finds files without the match, so we
+    # ignore the return code.
+    result = utils.run_cmd(cmd, no_shell=False, ignore_return_code=True)
+
+    error_msg = (
+        'Tests found in files without a "tests" module:\n {}'
+        'To ensure code coverage is reported correctly, please check that '
+        'your tests are in a module named "tests".'.format(result.stdout)
+    )
+
+    # stdout should be empty, where empty strings are false
```

Comment:
Please remove this comment or, if you want to keep it, rephrase it, something like this: `if stdout is not empty, the test should fail`. But I vote for removing it as it is pretty clear from the previous lines what you are asserting here (in my previous comment for this section I was removing it but it was probably hard to notice, sorry :D).

Reply:
Done and apologies,  I did miss that.